src/main/scala/org/broadinstitute/dsde/firecloud/service/LibraryService.scala

```diff
@@ -81,38 +121,65 @@ class LibraryService (protected val argUserInfo: UserInfo, val rawlsDAO: RawlsDA
           case Failure(e) =>
             Future(RequestCompleteWithErrorReport(BadRequest, BadRequest.defaultMessage, e))
           case Success(x) => {
-            rawlsDAO.getWorkspace(ns, name) flatMap { workspaceResponse =>
-              // this is technically vulnerable to a race condition in which the workspace attributes have changed
-              // between the time we retrieved them and here, where we update them.
-              val allOperations = generateAttributeOperations(workspaceResponse.workspace.attributes, userAttrs,
-                k => k.namespace == AttributeName.libraryNamespace && k.name != LibraryService.publishedFlag.name)
-              val republish = workspaceResponse.workspace.attributes.get(publishedFlag).fold(false)(_.asInstanceOf[AttributeBoolean].value)
-              rawlsDAO.patchWorkspaceAttributes(ns, name, allOperations) map { newws =>
-                if (republish) {
-                  // we do not need to delete before republish
-                  publishDocument(newws)
-                }
-                RequestComplete(newws)
+            rawlsDAO.getWorkspace(ns, name) map  { workspaceResponse =>
+              if (workspaceResponse.accessLevel >= WorkspaceAccessLevels.Write) {
```

Comment:
you shouldn't need this check; Rawls handles it. Is there a reason to enforce it here? Are you getting a head start on handling catalog permission?